central/resourcecollection/service/service_impl.go

```diff
@@ -134,6 +153,29 @@ func (s *serviceImpl) CreateCollection(ctx context.Context, request *v1.CreateCo
 	return &v1.CreateCollectionResponse{Collection: collection}, nil
 }
 
+func (s *serviceImpl) UpdateCollection(ctx context.Context, request *v1.UpdateCollectionRequest) (*v1.UpdateCollectionResponse, error) {
+	if !features.ObjectCollections.Enabled() {
+		return nil, errors.Errorf("%s env var is not enabled", features.ObjectCollections.EnvVar())
+	}
+
+	if request.GetId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Non empty collection id must be specified to delete a collection")
+	}
+
+	collection, err := collectionRequestToCollection(ctx, request, false)
+	if err != nil {
+		return nil, err
+	}
+	collection.Id = request.GetId()
```

Comment:
`collectionRequestToCollection` takes in a common interface `collectionRequest` for CreateCollectionRequest and UpdateCollectionRequest. Since Id is not a field of CreateCollectionRequest, that method can't be added to `collectionRequest` interface. I would revert to original idea of passing in an `idGetter` func argument for updates.

Reply:
Currently, the `collectionRequest` does not represent any collection request, it represents one type of request quite contradictory to the interface.

There are a couple of ways you can resolve, I will list a few:
**1. By changing the interface**
- Change the interface to:
```
type collectionRequest interface {
	GetID() string
	GetName() string
	GetDescription() string
	GetResourceSelectors() []*storage.ResourceSelector
	GetEmbeddedCollectionIds() []string
}
```
- Implement a wrapper for create request:
```
type createRequestWrapper struct {
	*v1.CreateCollectionRequest
}

func (*createRequestWrapper) GetID() string { return "" }

```
- Wrap the create request in the func call:
```
collection, err := collectionRequestToCollection(ctx, &createRequestWrapper{request}, true)
```
**2. Change the func signature** 
-  Change the function signature to receive `proto.Message` interface `collectionRequestToCollection(ctx context.Context, msg proto.Message)`. It removes dependency on `isCreate`
- Cast the incoming request:
```
if request, ok := msg.(*v1.CreateCollectionRequest); ok {
// do something
} else if request, ok := msg.(*v1.UpdateCollectionRequest); ok {
// do something
}
```